test/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests/dockerfile

```diff
@@ -0,0 +1,25 @@
+﻿# Create a container for running the OpenTelemetry redis unit tests.
+# This should be run from the root of the repo:
+#  opentelemetry>docker build -f test\OpenTelemetry.Instrumentation.StackExchangeRedis.Tests\dockerfile .
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS base
+
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1 as build
+ARG PUBLISH_CONFIGURATION=Release
+ARG PUBLISH_FRAMEWORK=netcoreapp3.1
+ARG TEST_FILTER=CategoryName!=RedisIntegrationTests
+WORKDIR /src
+COPY ["NuGet.config", ""] # Needed for the .NET 5 preview packages. Won't be needed in the future.
+COPY ["test/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests.csproj", "test/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests/"]
+COPY ["src/OpenTelemetry.Instrumentation.StackExchangeRedis/OpenTelemetry.Instrumentation.StackExchangeRedis.csproj", "src/OpenTelemetry.Instrumentation.StackExchangeRedis/"]
+COPY ["src/OpenTelemetry/OpenTelemetry.csproj", "src/OpenTelemetry/"]
+COPY ["src/OpenTelemetry.Api/OpenTelemetry.Api.csproj", "src/OpenTelemetry.Api/"]
+RUN dotnet restore "test/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests.csproj" --configfile "NuGet.config"
+COPY . .
+WORKDIR "/src/test/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests"
+RUN dotnet publish "OpenTelemetry.Instrumentation.StackExchangeRedis.Tests.csproj" -c "${PUBLISH_CONFIGURATION}" -f "${PUBLISH_FRAMEWORK}" -o /build -p:IntegrationBuild=true --no-restore
+RUN dotnet test "/build/OpenTelemetry.Instrumentation.StackExchangeRedis.Tests.dll" --filter "${TEST_FILTER}"
+
+FROM base AS final
+WORKDIR /test
+COPY --from=build /build .
+ENTRYPOINT ["dotnet", "test", "OpenTelemetry.Instrumentation.StackExchangeRedis.Tests.dll"]
```

Comment:
can we use docker strictly to spinup the dependencies (sql, redis), and use regular mechanism to run the actual tests? Just curious why you picked this approach of running tests from the docker itself.

Reply:
I just removed the non-integration test run from dockerfile.

Reply:
I was thinking more like - running *every* test using the usual way not inside the docker.  Docker just spins up a running Redis instance, and that's all. Integration tests Setup phase may validate that docker is running , setup ENV variable etc. ? 
Right now the actual test execution is also run *inside* the container, which is not really required right?

(I am happy to continue this conv outside of this PR, and let this merge first)